src/r2mm/manager/SettingsDexieStore.ts

```diff
@@ -130,19 +111,51 @@ export default class SettingsDexieStore extends Dexie {
 
     public async save(holder: ManagerSettingsInterfaceHolder) {
         const update = async () => {
+            // Update global settings.
             await this.global.toArray().then(async result => {
                 for (let settingsInterface of result) {
                     await this.global.update(settingsInterface.id!, {settings: JSON.stringify(holder.global)});
                 }
             });
-            await this.gameSpecific.toArray().then(async result => {
-                for (let settingsInterface of result) {
-                    await this.gameSpecific.update(settingsInterface.id!, {settings: JSON.stringify(holder.gameSpecific)});
-                }
-            });
+
+            // Update the active game's settings.
+            await this.games.put({ identifier: this.activeGame.settingsIdentifier, settings: JSON.stringify(holder.gameSpecific) });
         }
 
-        await this.transaction("rw!", this.global, this.gameSpecific, update);
+        await this.transaction("rw!", this.global, this.games, update);
+    }
+
+    /**
+     * Migrate the current game's legacy (v2) table to the v3 schema, if applicable.
+     * 
+     * v2 (whose types we still utilize) stored game-specific settings in individual tables.
+     * This worked but required us to bump the Dexie store version every time we added a new game. Yuck!
+     * To get around this we have moved all game settings into the `games` table, where the key
+     * is the settings identifier of the game, and the value is a JSON string.
+     * 
+     * @returns True if the legacy game field was migrated, false if it was not.
+     */
+    private async applyV2ToV3MigrationPlease(): Promise<Boolean> {
+        const gameTable = this.tables.find(table => table.name == this.activeGame.settingsIdentifier);
+        if (gameTable == undefined) {
+            return false;
+        }
+
+        const transaction = async () => {
+            const legacySettings = await gameTable.toCollection().first();
```

Comment:
Isn't this a read operation and therefore not really need to be inside the transaction? I.e. is the transaction needed at all if it's just one `put` operation?